### PR TITLE
added implementation of the getStatistic method

### PR DESCRIPTION
**Pull request by mstefanchuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = ",";
+    private static final String RESULT_HEADER = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/WorkWithFile.java:13]: https://github.com/mate-academy/jv-work-with-file/blob/master/checklist.md#divide-program-logic

Please split method logic to separate private methods (read file content, form report from file content and write report to file) and use these methods in the public one

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,66 @@
...
+
 public class WorkWithFile {
+    public static final String SEPARATOR = ",";
+    public static final String NEW_LINE = System.lineSeparator();
+    public static final String RESULT_HEADER = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+                BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:17]: move try catch finally into read, write methods
always include filename in exception messages related to files

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,66 @@
...
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        StringBuilder reportBuilder = new StringBuilder();
+
+        String[] lines = fileContent.split(NEW_LINE);
+        for (String line : lines) {
+            String[] parts = line.split(SEPARATOR);
+            if (parts.length == 2) {
+                String operation = parts[0];
+                int amount = Integer.parseInt(parts[1]);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:46]: extract OPERTAION_INDEX constant, etc

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,75 @@
...
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String fileContent = readFileContent(fromFileName);
+            String report = formReport(fileContent);
+            writeReportToFile(toFileName, report);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from or writing to file: "
+                    + e.getMessage(), e);
+        }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]: You handle exceptions within private methods, no need to additionally do it in parent method
```suggestion
            String fileContent = readFileContent(fromFileName);
            String report = formReport(fileContent);
            writeReportToFile(toFileName, report);
```

**Review by olekskov — APPROVED** [review]
Good job, approved.
Please pay attention to comment below for future

**Pull request closed without merging**